thundra/wrappers/fastapi/middleware.py

```diff
@@ -0,0 +1,142 @@
+from thundra.wrappers.fastapi.fastapi_wrapper import FastapiWrapper, logger
+from thundra.config import config_names
+
+from starlette.requests import Request
+from starlette.background import BackgroundTask
+
+from thundra.context.execution_context_manager import ExecutionContextManager
+from thundra.wrappers.fastapi.fastapi_utils import extract_headers
+
+from thundra.config.config_provider import ConfigProvider
+from thundra import constants
+
+import traceback, json, time
+
+import asyncio
+
+RESPONSE_REDIRECT_STATUS_CODE = 307
+
+def handle_error(exception, exception_handler):
+    error = {
+        'type': type(exception).__name__,
+        'message': str(exception),
+        'traceback': traceback.format_exc()
+    }
+    exception_handler(error)
+
+class ThundraMiddleware(object):
+    def __init__(self, app, opts=None):
+        self.app = app
+        self.opts = opts
+        self._wrapper = FastapiWrapper()
+
+    async def __call__(self, scope, receive, send):
+        """ASGI Middleware
+
+        Args:
+            scope (Dict): Connection information
+            receive (Callable): Get request from client
+            send (Callable): Send response to client
+            
+        """
+        if scope["type"] != "http":
+            return await self.app(scope, receive, send)
+
+        try:
+            self._wrapper.before_request(scope, None)
```

Comment:
Is it possible to inject `ExecutionContext`, which is returned by `_wrapper.before_request`, into scope or request and get the `ExecutionContext` from there below. My point is that if we are able to get the `ExecutionContext` from the scope or request directly, in case of possible context propagation issues, we will always finish the execution context

Reply:
done.